pkgs/development/interpreters/bwbasic/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, dos2unix, fetchzip }:
+
+stdenv.mkDerivation rec {
+  pname = "bwbasic";
+  version = "3.20";
+
+  src = fetchzip {
+    url = "mirror://sourceforge/project/bwbasic/bwbasic/version%203.20/bwbasic-3.20.zip";
+    extension = "zip";
```

Comment:
The ".zip" extension is now detectable from url.

Also, you don't need to use `fetchzip`, you can use `fetchurl` and put `unzip` in `nativeBuildInputs`. The benefit of using fetchurl over fetchzip is that the https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/copy-tarballs.pl supports backing up those files to tarballs.nixos.org.

Reply:
Ah thanks! Good to know. I am in favor of backups, for sure, so I've changed that.

Do you know if it's possible to make this work without writing my own unpackPhase (as the latest revision does)?